pkg/gce-pd-csi-driver/controller.go

```diff
@@ -224,14 +224,16 @@ func (gceCS *GCEControllerServer) DeleteVolume(ctx context.Context, req *csi.Del
 
 	volKey, err := common.VolumeIDToKey(volumeID)
 	if err != nil {
-		// Cannot find volume associated with this ID because can't even get the name or zone
-		// This is a success according to the spec
+		// Cannot find volume associated with this ID because VolumeID is not in
+		// correct format, this is a success according to the Spec
+		klog.Warningf("Treating volume as deleted because volume id %s is invalid: %v", volumeID, err)
```

Comment:
Should it return success? Can we get into a case where user pre-provisions a PV, but gets the volume id wrong.  Then when they go and delete it they think it's successful, when it's actually not, and then we leak a volume.

Reply:
I don't agree with the return code in this case for many reasons, but this is what it says in the Spec as well as CSI Sanity.